### PR TITLE
fix a few interop test issues

### PR DESCRIPTION
**Pull request by Stebalien**
The _interop_ tests passed but the interface tests didn't.

* Return "not a directory" again when trying to cat a directory.
* Use EmitOnce in `config`.
* Return the correct error in `bootstrap rm`.

Other issues:

* We now disagree on size: https://github.com/ipfs/interface-js-ipfs-core/issues/427
* We now normalize paths on add: https://github.com/ipfs/interface-js-ipfs-core/pull/440

### CODE REVIEW

File: core/commands/cat.go [hunk 1]
```diff
@@ -128,8 +128,14 @@ func cat(ctx context.Context, api iface.CoreAPI, paths []string, offset int64, m
...
-		file, ok := f.(files.File)
-		if !ok {
+		var file files.File
+		switch f := f.(type) {
+		case files.File:
+			file = f
+		case files.Directory:
+			return nil, 0, iface.ErrIsDir
+		case *files.Symlink:
+			// TODO: Consider returning "error not supported"?
```
Comment (Stebalien) [inline core/commands/cat.go]: We can handle this later, IMO.

File: core/commands/cat.go [hunk 2]
```diff
@@ -128,8 +128,14 @@ func cat(ctx context.Context, api iface.CoreAPI, paths []string, offset int64, m
...
-		file, ok := f.(files.File)
-		if !ok {
+		var file files.File
+		switch f := f.(type) {
+		case files.File:
+			file = f
+		case files.Directory:
+			return nil, 0, iface.ErrIsDir
+		default:
+			// TODO: Consider returning "error not supported" for symlinks.
```
Comment (magik6k) [inline core/commands/cat.go]: Doesn't go-unixfs do that? https://github.com/ipfs/go-unixfs/blob/master/io/dagreader.go#L80

Comment (Stebalien) [reply core/commands/cat.go]: Yeah, I just didn't feel like adding that error to the core API yet (I'm being lazy and trying to cut a release...).

Comment (Stebalien) [reply core/commands/cat.go]: @magik6k https://github.com/ipfs/interface-go-ipfs-core/pull/5

**Comment by Stebalien** [comment]
Travis is failing due to https://github.com/ipfs/go-ipfs-cmds/issues/143 (I think).

**Pull request merged**